ikarus/utils/concepts.hh

```diff
@@ -198,6 +201,19 @@ namespace Concepts {
     { adaptiveStepSizing.setTargetIterations(std::declval<int>()) } -> std::same_as<void>;
   };
 
+  /**
+   * \concept Observable
+   * \brief Concept to check if the type of state passed to an observable is valid or not.
+   *
+   * \tparam MT The type of message that the observable can handle.
+   * \tparam ST The type of the state of the observable.
+   */
+  template <typename MT, typename ST>
+  concept Observable = requires {
+    (std::is_same_v<MT, ControlMessages> && std::is_same_v<ST, ControlState>) ||
+        (std::is_same_v<MT, NonLinearSolverMessages> && std::is_same_v<ST, NonLinearSolverState>);
```

Comment:
This concept does not make sense, since it is not scalable. You should check, if `MT` and `ST` adhere another concept and not check that they are of a specific type.